src/dbnode/storage/entry.go

```diff
@@ -141,14 +141,52 @@ func (entry *Entry) IndexedForBlockStart(indexBlockStart xtime.UnixNano) bool {
 	return isIndexed
 }
 
-// IndexedRange returns minimum and maximum blockStart values covered by index entry.
+// ReconciledIndexedForBlockStart returns true if the blockStart has been indexed.
+// Attempts to retrieve the most recent index entry from the shard if this entry
+// was never inserted there. If there is an error, returns the value for this entry.
+// Additionally, returns a boolean to denote if result came from reconciliation.
+func (entry *Entry) ReconciledIndexedForBlockStart(indexBlockStart xtime.UnixNano) (bool, bool) {
```

Comment:
The prefixing of these methods to "Reconciled..." is great since it tells you what it's doing.

Perhaps we should do the same for `TryMarkIndexGarbageCollected` and have that renamed for consistency to `ReconciledTryMarkIndexGarbageCollected`?

Reply:
Also I wonder since we check `if entry.insertTime.Load() > 0 {` here for both of these methods whether we should do the same for `ReconciledTryMarkIndexGarbageCollected` to avoid needing to acquire the lock in the regular case there too.

Reply:
Since we now have a single method to return a reconciled entry, going to leave the rest of these methods named as-is